upi/openstack/03_bootstrap.yaml

```diff
@@ -0,0 +1,35 @@
+# Required Python packages:
+#
+# ansible
+# openstacksdk
+# netaddr
+
+- hosts: all
+
+  vars:
+    bootstrap_data: "{{ os_infra_id }}-bootstrap-ignition.json"
+
+    bootstrap_server: "{{ os_infra_id }}-bootstrap"
+    bootstrap_port: "{{ os_infra_id }}-bootstrap-port"
+
+  tasks:
+  - name: 'Create the bootstrap server port'
+    os_port:
+      name: "{{ bootstrap_port }}"
+      network: "{{ os_network }}"
+      security_groups:
+      - "{{ os_sg_master }}"
+      allowed_address_pairs:
+      - ip_address: "{{ os_subnet_range | next_nth_usable(5) }}"
+      - ip_address: "{{ os_subnet_range | next_nth_usable(6) }}"
+      - ip_address: "{{ os_subnet_range | next_nth_usable(7) }}"
+
+  - name: 'Create the bootstrap server'
+    os_server:
+      name: "{{ bootstrap_server }}"
+      image: "{{ os_image_rhcos }}"
+      flavor: "{{ os_flavor_master }}"
+      userdata: "{{ lookup('file', bootstrap_data) }}"
```

Comment:
When I tried to run this, Ansible printed out this warning:

```
TASK [Create the bootstrap server] ************************************************************************************************************************************************************
[WARNING]: The value {'passwd': {}, 'networkd': {}, 'storage': {}, 'systemd': {}, 'ignition': {'timeouts': {}, 'security': {}, 'version': '2.2.0', 'config': {'append': [{'source':
'http://pastebin.test.redhat.com/pastebin.php?dl=817154', 'verification': {}}]}}} (type dict) in a string field was converted to u"{'passwd': {}, 'networkd': {}, 'storage': {}, 'systemd':
{}, 'ignition': {'timeouts': {}, 'security': {}, 'version': '2.2.0', 'config': {'append': [{'source': 'http://example.com/817154', 'verification': {}}]}}}" (type
string). If this does not look like what you expect, quote the entire value to ensure it does not change.
```

So lookup parsed the file as json, converted it to a python dictionary and then Ansible converted it back to a string -- but messed it up. Looks like it just converted it to a stringified dict rather than json (all the quotes are wrong).

And indeed the server failed to boot due to that.

I've tried to convert the `lookup` to a string explicitly and that seems to have done the trick:

    userdata: "{{ lookup('file', bootstrap_data) | string }}"

So again, this will need to be fixed in all the `os_server` calls.

Reply:
thank you!